src/read_ic.c

```diff
@@ -1,1342 +1,1384 @@
+#ifdef RADTRANSFER
+            for(n = 0; n < pc; n++) {for(k = 0; k < N_RT_FREQ_BINS; k++) {SphP[offset + n].Rad_E_gamma[k] = *fp++;}}
+#endif
+            break;
+
+            /* adaptive softening parameters */
+        case IO_AGS_SOFT:
+#if defined (AGS_HSML_CALCULATION_IS_ACTIVE) && defined(AGS_OUTPUTGRAVSOFT)
+            for(n = 0; n < pc; n++) {PPP[offset + n].AGS_Hsml = *fp++;}
+#endif
+            break;
+
+        case IO_AGS_ZETA:
+#if defined (AGS_HSML_CALCULATION_IS_ACTIVE) && defined(AGS_OUTPUTZETA)
+            for(n = 0; n < pc; n++) {PPPZ[offset + n].AGS_Zeta = *fp++;}
+#endif
+            break;
+
+        case IO_CHIMES_ABUNDANCES:
+#if defined(CHIMES) && !defined(CHIMES_INITIALISE_IN_EQM)
+            for (n = 0; n < pc; n++)
+            {
+    	        allocate_gas_abundances_memory(&(ChimesGasVars[offset + n]), &ChimesGlobalVars);
+	            for (k = 0; k < ChimesGlobalVars.totalNumberOfSpecies; k++) {ChimesGasVars[offset + n].abundances[k] = (ChimesFloat) (*fp++);}
+#ifdef CHIMES_TURB_DIFF_IONS
+                chimes_update_turbulent_abundances(n, 1);
+#endif
+            }
+#endif
+            break;
+
+        case IO_COSMICRAY_ENERGY:
+            break;
+
+        case IO_COSMICRAY_SLOPES:
+            break;
+
+        case IO_COSMICRAY_ALFVEN:
+            break;
+
+        case IO_OSTAR:
+#ifdef GALSF_SFR_IMF_SAMPLING
+             for(n = 0; n < pc; n++) {P[offset + n].IMF_NumMassiveStars = *fp++;}
+#endif
+            break;
+            
+        case IO_UNSPMASS:
+#if defined(BH_WIND_SPAWN) && defined(BH_DEBUG_SPAWN_JET_TEST)
+             for(n = 0; n < pc; n++) {P[offset + n].unspawned_wind_mass = *fp++;}
+#endif
+            break; 
+            
+        case IO_IDEN:
+#if defined(SPAWN_B_POL_TOR_SET_IN_PARAMS) && defined(BH_DEBUG_SPAWN_JET_TEST)
+             for(n = 0; n < pc; n++) {SphP[offset + n].IniDen = *fp++;}
+#endif
+            break;
+            
+        case IO_INIB:        
+#if defined(SPAWN_B_POL_TOR_SET_IN_PARAMS) && defined(BH_DEBUG_SPAWN_JET_TEST)
+             for(n = 0; n < pc; n++) {for(k=0;k<3;k++) {SphP[offset + n].IniB[k] = *fp++;}}
+#endif      
+            break;  
+            
+        case IO_TURB_DYNAMIC_COEFF:
+#ifdef TURB_DIFF_DYNAMIC
+            for (n = 0; n < pc; n++) {SphP[offset + n].TD_DynDiffCoeff = *fp++;}
+#endif
+            break;
+
+        case IO_SINKRAD:
+#ifdef BH_GRAVCAPTURE_FIXEDSINKRADIUS
+            for(n = 0; n < pc; n++) {P[offset + n].SinkRadius = *fp++;}
+#endif
+            break;
+            
+        case IO_MOLECULARFRACTION:
+#if defined(COOL_MOLECFRAC_NONEQM) & !defined(IO_MOLECFRAC_NOT_IN_ICFILE)
+            for (n = 0; n < pc; n++) {SphP[offset + n].MolecularMassFraction_perNeutralH = *fp++;}
+#endif
+            break;
+
+        /* SLUG objects*/
+        case IO_SLUG_STATE_RNG:  /* It may not be *ip_int since it is a unit128 type */
+            for(n = 0; n < pc; n++) {P[offset + n].rngStateAtBirth = *ip_int++;}

```

Comment:
You will have to modify this to use the bitshift operator to combine the lower and upper 64 bits of the full 128-bit integer.